features/api/plan_features.feature

```diff
@@ -0,0 +1,75 @@
+Feature: Plan Features
+  In order to list which features of my API are available for each plan
+  As a provider
+  I want to define features
+
+  Background:
+    Given a provider "foo.3scale.localhost"
+    And an application plan "Basic" of provider "foo.3scale.localhost"
+    Given current domain is the admin domain of provider "foo.3scale.localhost"
+
+  @javascript
+  Scenario: Create a feature
+    When I log in as provider "foo.3scale.localhost"
+    And I go to the edit page for plan "Basic"
+    And I follow "New feature"
+    And I fill in "Name" with "Free T-shirt"
+    And I fill in "System name" with "tee"
+    And I fill in "Description" with "T-shirt with logo of our company for free."
+    And I press "Save"
+    Then I should see "T-shirt with logo of our company for free."
+    And I should see the flash message "Feature has been created."
+
+  @javascript
+  Scenario: "No features yet" notice disappears when the first feature is created, and appears when last one is deleted.
+    When I log in as provider "foo.3scale.localhost"
+    And I go to the edit page for plan "Basic"
+    Then I should see "This plan has no features yet."
+
+    When application plan "Basic" has "Some Feature" enabled
+    And I go to the edit page for plan "Basic"
+    Then I should not see "This plan has no features yet."
+
+    When application plan "Basic" does not have any features
+    And I go to the edit page for plan "Basic"
+    Then I should see "This plan has no features yet."
+
+  @javascript
+  Scenario: Disable a feature
+    Given application plan "Basic" has "Some Feature" enabled
+    And I log in as provider "foo.3scale.localhost"
+    And I go to the edit page for plan "Basic"
```

Comment:
Half steps are repeated and could be moved into a Background. The feature being enabled/disabled can be handled by using 2 different Rule.

Reply:
Sorry, I'm not sure what you mean here.

Reply:
@josemigallas `I log in as provider "foo.3scale.localhost"` and `I go to the edit page for plan "Basic"` appear in all scenarios, but only the "I login" step could be moved to the background. The "go to edit" step only makes sense after the plan has been properly configured, which is different in every scenario, so that step can't be moved to the background. About using 2 different Rule, I don't know what you mean neither.

Reply:
I moved both `log in` and `has "Some Feature" enabled` steps to the Background.
`I go to the edit page for plan "Basic"` is still repeated in all steps, but in 2 (out of 7) they are not the 1st step.

I guess that by Rule you might mean [Gherkin Rules](https://cucumber.io/docs/gherkin/reference/#rule). To be honest, I don't think it's semantically applicable in this case - there is no generic "business rule" that would be grouping here. I think adding them here just for extracting one step to background would make the scenarios more confusing, and IMO is not worth it.